choreolib/src/main/java/choreo/util/AllianceFlipUtil.java

```diff
@@ -226,4 +229,31 @@ public static Rotation3d flip(Rotation3d rotation) {
   public static Pose3d flip(Pose3d pose) {
     return new Pose3d(flip(pose.getTranslation()), flip(pose.getRotation()));
   }
+
+  /**
+   * Creates a Supplier<Optional<Pose2d>> based on a Supplier<Optional<Alliance>> and original Optional<Pose2d>
+   * @param pose The pose to flip
+   * @param alliance The current alliance
+   * @param doFlip Returns true if flipping based on the alliance should be done
+   * @return empty if the alliance is empty;
+   *  the original pose if the alliance is blue or doFlip is false;
+   *  the flipped pose if the alliance is red and doFlip is true
+   */
+  public static Supplier<Optional<Pose2d>> optionalFlipped(Optional<Pose2d> poseOpt, Supplier<Optional<Alliance>> allianceOpt, BooleanSupplier doFlip) {
+    return ()->allianceOpt.get().flatMap(
```

Comment:
Ditto, implementation should be `() -> doFlip.getAsBoolean() ? allianceOpt.get().flatMap(ally -> poseOpt.map(pose -> ally == Alliance.Red ? flip(pose) : pose)) : pose;`